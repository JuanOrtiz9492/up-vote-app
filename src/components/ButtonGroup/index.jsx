import React, {Component} from 'react';
import styled from 'styled-components';
import Button from '../Button';
import {ThumbsUpIcon, ThumbsDownIcon} from '../Icon/IconList'
import { cssConstants } from '../../config/cssConstants';

const VotesContainer = styled.div`
    display: flex;
    justify-content: space-around;
    flex-direction: row;
    width: 15em;
    margin: 1em 0;
`

class ButtonGroup extends Component {
    constructor(props){
        super(props);
        this.state = {
            isUpVoteSelected: false,
            isDownVoteSelected: false,
            isVoteSend: false
        }
    }
    handleUpVote = ()=> {
        this.setState({
            isUpVoteSelected: true,
            isDownVoteSelected: false,
        });
    }

    handleDownVote = ()=> {
        this.setState({
            isUpVoteSelected: false,
            isDownVoteSelected: true,
        });
    }

    handleVoteSend = () => {
        const { isUpVoteSelected, isDownVoteSelected } = this.state;
        const { voteHandler } = this.props;
        if(isUpVoteSelected) {
            voteHandler('upVote')
        } else if (isDownVoteSelected) {
            voteHandler('downVote')
        }

        if(isUpVoteSelected || isDownVoteSelected){
            this.setState({
                isVoteSend: true,
                isUpVoteSelected: false,
                isDownVoteSelected: false,
            })
            this.props.setIsVoteSend(true);
        } else {
            this.setState({isVoteSend: false})
            this.props.setIsVoteSend(false);
        }
    }
    
    render () {
        const {
            isUpVoteSelected,
            isDownVoteSelected,
            isVoteSend,
        } = this.state;
        const { colors } = cssConstants;
        return (
            <>
            <VotesContainer>
                {!isVoteSend && 
                <>
                    <Button 
                        jungle 
                        icon={<ThumbsUpIcon width={1.2} color={colors.white.hex}/>} 
                        checked={isUpVoteSelected} 
                        onClick={this.handleUpVote}
                    />
                    <Button 
                        fire 
                        icon={<ThumbsDownIcon width={1.2} color={colors.white.hex}/>} 
                        checked={isDownVoteSelected} 
                        onClick={this.handleDownVote}
                    />
                </>}
                <Button label={isVoteSend ? "vote again" : "vote now"} onClick={this.handleVoteSend}/>
            </VotesContainer>
            </>
        )
    }
}

export default ButtonGroup;
