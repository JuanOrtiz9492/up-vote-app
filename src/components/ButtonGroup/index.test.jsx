import React from 'react';
import { render } from '@testing-library/react';
import ButtonGroup from '.';

describe('ButtonGroup Component', () => {

    it('should Match the Snapshot', () => {
        const { container } = render(<ButtonGroup />);
        expect(container).toMatchSnapshot();
      })
})

