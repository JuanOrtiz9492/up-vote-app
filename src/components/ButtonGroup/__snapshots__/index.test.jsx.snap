// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Component should Match the Snapshot 1`] = `
<div>
  <div
    class="sc-AykKJ iAKuBh"
  >
    <button
      class="sc-AykKC dEOuyc jungle"
      color="#ffffff"
    >
      <svg
        class="sc-AykKD SFEtK"
        color="#ffffff"
        width="1.2"
      >
        thumbs-up.svg
      </svg>
    </button>
    <button
      class="sc-AykKC dEOuyc fire"
      color="#ffffff"
    >
      <svg
        class="sc-AykKE kfwLkP"
        color="#ffffff"
        width="1.2"
      >
        thumbs-down.svg
      </svg>
    </button>
    <button
      class="sc-AykKC dEOuyc"
      color="#ffffff"
    >
      vote now
    </button>
  </div>
</div>
`;
